e2e/cypress/integration/messaging/emoji_keyboard_entry_spec.js

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// [number] indicates a test step (e.g. 1. Go to a page)
+// [*] indicates an assertion (e.g. * Check the title)
+// Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Message', () => {
+    beforeEach(() => {
+        // # Login as "user-1" and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M16738 Use keyboard navigation in emoji picker', () => {
+        // # Open emoji picker
+        cy.get('#emojiPickerButton').click();
+
+        // # Wait for emojis to load
+        cy.get('#emojiPicker').should('be.visible');
+
+        // # Move around and verify emojis with directional arrow input
+        const pressCounts = {
+            right: 5,
+            down: 5,
+            left: 4,
+            up: 5,
+        };
+        for (const direction of Object.keys(pressCounts)) {
```

Comment:
I think we should still be even more explicit with what emoji we expect to be on. If press down or left once, we should hit the first emoji, but what if we arrow key and the selection does not move? With this current code, we just verify that the selected image is the one previewed, so it would still pass.

Given our know state (or arrangement) of emojis, I think we should give input and expect to be on a specific emoji, then verify that it has the selected class and that the preview is correct.

Reply:
ee20a82a260e1ec4e200ddbbe9adf9205e18d890 should resolve this concern to some extent.  It now determines that the selected emoji _has_ changed but not that it moved to the _correct_ emoji.  This is difficult to predetermine for up/down arrow presses along with a potentially variable width popup.